py/desispec/spectra.py

```diff
@@ -658,6 +666,168 @@ def update(self, other):
 
         return
 
+    def to_specutils(self):
+        """Convert to ``specutils`` objects.
+
+        Returns
+        -------
+        specutils.SpectrumList
+            A list with each band represented as a specutils.Spectrum1D object.
+
+        Raises
+        ------
+        NameError
+            If ``specutils`` is not available in the environment.
+        """
+        if not _specutils_imported:
+            raise NameError("specutils is not available in the environment.")
+        sl = SpectrumList()
+        AA = Unit('Angstrom')
+        specunit = Unit('10-17 erg cm-2 s-1 AA-1')
+        for i, band in enumerate(self.bands):
+            meta = {'band': band}
+            spectral_axis = self.wave[band] * AA
+            flux = self.flux[band] * specunit
+            uncertainty = InverseVariance(self.ivar[band] * (specunit**-2))
+            mask = self.mask[band] != 0
+            meta['int_mask'] = self.mask[band]
+            meta['resolution_data'] = self.resolution_data[band]
+            try:
+                meta['extra'] = self.extra[band]
+            except KeyError:
+                meta['extra'] = None
+            if i == 0:
+                #
+                # Only add these to the first item in the list.
+                #
+                meta['bands'] = self.bands
+                meta['fibermap'] = self.fibermap
+                meta['exp_fibermap'] = self.exp_fibermap
+                meta['desi_meta'] = self.meta
+                meta['single'] = self._single
+                meta['scores'] = self.scores
+                meta['scores_comments'] = self.scores_comments
+                meta['extra_catalog'] = self.extra_catalog
+            sl.append(Spectrum1D(flux=flux, spectral_axis=spectral_axis,
+                                 uncertainty=uncertainty, mask=mask, meta=meta))
+        return sl
+
+    @classmethod
+    def from_specutils(cls, spectra):
+        """Convert ``specutils`` objects to a :class:`~desiutil.spectra.Spectra` object.
+
+        Parameters
+        ----------
+        spectra : specutils.Spectrum1D or specutils.SpectrumList
+            A ``specutils`` object.
+
+        Returns
+        -------
+        :class:`~desiutil.spectra.Spectra`
+            The corresponding DESI-internal object.
+
+        Raises
+        ------
+        NameError
+            If ``specutils`` is not available in the environment.
+        ValueError
+            If an unknown type is found in `spectra`.
+        """
+        if not _specutils_imported:
+            raise NameError("specutils is not available in the environment.")
+        if isinstance(spectra, SpectrumList):
+            sl = spectra
+            try:
+                bands = sl[0].meta['bands']
+            except KeyError:
+                #
+                # This is a big assumption; it doesn't capture ['b', 'z'] or ['r', 'z'].
+                #
+                bands = ['b', 'r', 'z'][0:len(sl)]
```

Comment:
@akremin's suggestion above is maybe to simply raise an exception if it is not possible to determine the bands. If I can quickly figure out a wavelength range solution, that's great, but if it takes too much time, I will fall back to an exception.

Reply:
Based on discussions today, @sbailey & I agree that when converting `specutils` to DESI internal, users really *should* supply band information, and raise an error if they do not.  It would be supplied via `my_specutils_object.meta['bands']`, so that's not hard.